### PR TITLE
Fetch CA certificate tests

### PR DESCRIPTION
**Pull request by tomasz-stefaniak**
## Description

[ What changed? Feel free to be brief. ]

## Checklist

- [] I've read the [contributing guide](https://github.com/continuedev/continue/blob/main/CONTRIBUTING.md)
- [] The relevant docs, if any, have been updated or created
- [] The relevant tests, if any, have been updated or created

## Screen recording or screenshot

[ When applicable, please include a short screen recording or screenshot - this makes it much easier for us as contributors to review and understand your changes. See [this PR](https://github.com/continuedev/continue/pull/6455) as a good example. ]

## Tests

[ What tests were added or updated to ensure the changes work as expected? ]

    

---

## Summary by cubic
Added comprehensive tests for SSL certificate handling in the fetch package and updated CI to run these tests on every PR.

- **Testing**
  - Added tests for custom CA bundles, data URI certificates, SSL config options, caching, and error handling.
  - Improved test coverage for certificate-related edge cases.
- **CI**
  - Updated workflow to install fetch dependencies and run fetch package tests during PR checks.

### CODE REVIEW

File: .github/workflows/pr_checks.yaml [hunk 1]
```diff
@@ -609,6 +609,32 @@ jobs:
...
+      - name: Install fetch dependencies
+        if: steps.fetch-cache.outputs.cache-hit != 'true'
+        run: |
+          cd packages/fetch
+          npm ci
+
+      - name: Run fetch tests
+        run: |
+          cd packages/fetch
+          npm test
```
Comment (Patrick-Erichsen) [inline .github/workflows/pr_checks.yaml]:
```suggestion
       - name: Build packages and run tests
          run: |
            ./scripts/build-packages.sh
            cd packages/fetch
            npm test
```

I just modified a bunch of the logic in `pr_checks.yaml` to try and simplify things, eliminate copy/paste, use the same build script locally + in CI, etc. The above diff is what were doing for the other packages now, eg https://github.com/continuedev/continue/blob/main/.github/workflows/pr_checks.yaml#L422

The downside of this is that we don't have caching on the packages, but that was a bit difficult to set up, and also doesn't really matter since it takes like 30s so it's unlikely any of the package-related jobs will ever be the upper bounds of our PR runs.

 Tabs are a bit off in this diff though so don't commit directly.

**Comment by Patrick-Erichsen** [comment]
Will follow up on this once we get CI issues fixed

**Review by Patrick-Erichsen — CHANGES_REQUESTED** [review]
The changes to `.github/workflows/pr_checks.yaml` can be discarded, we now run tests on all packages by default: https://github.com/continuedev/continue/blob/main/.github/workflows/pr-checks.yaml#L178-L181
